lua/luasnip/util/environ.lua

```diff
@@ -1,190 +1,197 @@
-	return os.date("%m")
-end
-function lazy_vars.CURRENT_MONTH_NAME()
-	return os.date("%B")
-end
-function lazy_vars.CURRENT_MONTH_NAME_SHORT()
-	return os.date("%b")
-end
-function lazy_vars.CURRENT_DATE()
-	return os.date("%d")
-end
-function lazy_vars.CURRENT_DAY_NAME()
-	return os.date("%A")
-end
-function lazy_vars.CURRENT_DAY_NAME_SHORT()
-	return os.date("%a")
-end
-function lazy_vars.CURRENT_HOUR()
-	return os.date("%H")
-end
-function lazy_vars.CURRENT_MINUTE()
-	return os.date("%M")
-end
-function lazy_vars.CURRENT_SECOND()
-	return os.date("%S")
-end
-function lazy_vars.CURRENT_SECONDS_UNIX()
-	return tostring(os.time())
+	return o
 end
 
--- For inserting random values
+local builtin_ns_names = vim.inspect(vim.tbl_keys(builtin_vars.builtin_ns))
+
+function Environ.env_namespace(name, namespace)
+	assert(
+		#name > 0 and not (name:find("_")),
+		("You can't create a namespace with name '%s' empty nor containing _"):format(
+			name
+		)
+	)
+	assert(
+		not builtin_vars.builtin_ns[name],
+		("You can't create a namespace with name '%s' because is one one of %s"):format(
+			name,
+			builtin_ns_names
+		)
+	)
+	local ns = namespace
+
+	assert(
+		ns and type(ns) == "table",
+		("Your namespace '%s' has to be a table"):format(name)
+	)
+	assert(
+		ns.init or ns.vars,
+		("Your namespace '%s' needs init or vars"):format(name)
+	)
+
+	-- namespace.eager → ns.vars
+	assert(
+		not ns.eager or ns.vars,
+		("Your namespace %s can't set a `eager` field without the `vars` one"):format(
+			name
+		)
+	)
+
+	ns.eager = ns.eager or {}
+	local is_table = ns.is_table or false
+
+	local type_of_it = type(is_table)
+
+	assert(
+		type_of_it == "table"
+			or type_of_it == "boolean"
+			or type_of_it == "function",
+		("Your namespace %s can't have `is_table` of type %s"):format(
+			name,
+			type_of_it
+		)
+	)
+
+	-- If type is function we don't have to override it
+	if type_of_it == "table" then
+		local is_table_set = {}
+
+		for _, key in ipairs(is_table_set) do
+			is_table_set[key] = true
+		end
 
-math.randomseed(os.time())
+		ns.is_table = function(key)
+			return is_table_set[key] or false
+		end
+	elseif type_of_it == "boolean" then
+		return function(_)
+			return is_table
+		end
+	end
 
-function lazy_vars.RANDOM()
-	return string.format("%06d", math.random(999999))
-end
+	if ns.vars and type(ns.vars) == "table" then
+		ns.vars = tbl_to_lazy_env(ns.vars)
+	end
 
-function lazy_vars.RANDOM_HEX()
-	return string.format("%06x", math.random(16777216)) --16^6
+	namespaces[name] = ns
 end
 
-function lazy_vars.UUID()
-	local random = math.random
-	local template = "xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx"
-	local out
-	local function subs(c)
-		local v = (((c == "x") and random(0, 15)) or random(8, 11))
-		return string.format("%x", v)
+function Environ:__index(key)
+	local nmsp, varname = _resolve_namespace_var(key)
+	---@diagnostic disable-next-line: need-check-nil
+	local val = nmsp.vars(varname)
+
+	if val then
+		val = tostring(val)
```

Comment:
This could produce unwanted results if `val` is a table ("table 0xsomething"). Is the `tostring` even necessary? Imo it's the namespaces responsibility to actually just return string/string[].